File: back-end-maven/src/main/java/net/weg/wegssm/model/service/ResponsavelNegocioService.java
package net.weg.wegssm.model.service;

import lombok.AllArgsConstructor;
import net.weg.wegssm.model.entities.ResponsavelNegocio;
import net.weg.wegssm.repository.ResponsavelNegocioRepository;
import org.springframework.stereotype.Service;

import java.util.Optional;

/**
 * Classe service para os responsáveis de negócio
 */
@Service
@AllArgsConstructor
public class ResponsavelNegocioService {

    /**
     * Classe service dos responsáveis de negócio
     */
    private ResponsavelNegocioRepository responsavelNegocioRepository;

    /**
     * Função booleana para verificação se um responsável de negócio existe pelo seu ID
     *
     * @param id - Id do responsável de negócio
     * @return - Verificando se existe um responsável de negócio ou não
     */
    public Boolean existsById(Long id) {
        return responsavelNegocioRepository.existsById(id);
    }

    /**
     * Função para salvar um responsável negócio
     *
     * @param entity - Responsável negócio a ser salvo
     * @param <S>    - Responsável negócio
     * @return - Responsável negócio salvo
     */
    public <S extends ResponsavelNegocio> S save(S entity) {
        return responsavelNegocioRepository.save(entity);
    }

    /**
     * Função para excluir um responsável de negócio pelo seu ID
     *
     * @param id - Id do responsável de negócio
     */
    public void deleteById(Long id) {
        responsavelNegocioRepository.deleteById(id);
    }

    /**
     * Função para buscar um responsável negócio pelo seu ID
     *
     * @param id - Id do responsável negócio
     * @return - Responsável negócio encontrado
     */
    public Optional<ResponsavelNegocio> findById(Long id) {
        return responsavelNegocioRepository.findById(id);
    }

}

File: front-end/src/service/responsavelNegocioService.js
import axios from "./api";

/** Classe service para os responsáveis de negócio */
class ResponsavelNegocioService {

    /** Função para criar um novo responsável de negócio */
    async post(responsavelNegocio) {
        return (await axios.post(`/responsavel_negocio`, responsavelNegocio, { headers: { "Content-Type": "application/json" } })).data;
    }

    /** Função para atualizar um responsável de negócio */
    async put(responsavelNegocio) {
        return (await axios.put(`/responsavel_negocio`, responsavelNegocio, { headers: { "Content-Type": "application/json" } })).data;
    }

    /** Função para excluir um responsável de negócio pelo seu ID */
    async delete(responsavelNegocioId) {
        return (await axios.delete(`/responsavel_negocio/${responsavelNegocioId}`)).data;
    }

}

export default new ResponsavelNegocioService();